### DISCUSSION TITLE
Remote change discarded - not in --single-directory sync scope (not in DB)

### DISCUSSION THREAD

**Discussion by bobmclane** [abraunegg/onedrive#1308] [Q&A] [answered]
Hi
I am trying to sync 1 folder from opt/onedrive/folder1 which I can do successfully using upload-only, however I would like to sync both ways however I get the message "Remote change discarded - not in --single-directory sync scope (not in DB)" (is in verbose but I assume this is checking every file/folder even if I exclude them in the sync_lists file.. is this correct?
If so is there a way to get it to sync both ways but exclude everything else from a download of folder1 from OneDrive without it going thru every file to check/exclude it?

Thanks
Bob.!

**Comment by abraunegg** [comment, marked as answer]
@bobmclane 
Please read the usage document in regard to utilising the `--single-directory` option and do not use 'sync_list'

**Reply by bobmclane** [reply]
@abraunegg  - Hi thanks for replying so quickly... 
So I have tried this with the folder 'folder1' works perfectly with upload-only syncs a new file straight up no bother, however if I set to download or remove the upload argument it sits there for ages.  If I switch on the verbose argument I see lots of the "Remote change discarded - not in --single-directory sync scope (not in DB)" so this seems to me to be running thru all files in my OneDrive area which I don't want it to do, is that right? 

So me specifying the following command is this supposed to ignore all else and only download the contents of folder1 without checking to anything else?

**onedrive --synchronize --single-directory 'folder1'**

Sorry trying to understand how it works or if I've missed something (highly possible! :))

Thanks 
Bob

**Reply by abraunegg** [reply]
@bobmclane 
When using 'sync_list' - every OneDrive object has to be checked, because the API has been written in such a way that is assumes that all changes are made exactly to the local directory. By using 'sync_list' it prevents using the /delta API call correctly.

When using `--single-directory`, the /delta API call can be used correctly, however, the /delta call provides all changes made online since your last sync. All of those delta changes need to be checked against your command request of `--single-directory` thus this is why you see this message `Remote change discarded - not in --single-directory sync scope (not in DB)` when a remote change (made on OneDrive) does not fit your `--single-directory` criteria.

> onedrive --synchronize --single-directory 'folder1'

This is the correct command to use for bi-directional syncing of a single folder. To achieve the best results, do not use 'sync_list' when just syncing one single folder.

**Reply by bobmclane** [reply]
@abraunegg  - Thanks for the reply, sorry been a while since I looked at this... so final question I promise :) So  I think I have it working but if I have TB's of data in OneDrive and my "folder1" is in the root along with other files and folders in the root if I perform a --single-directory sync on "folder1" is it still checking every other file/folder in my OneDrive as it seems so? Also will it do this every time as this could take me ages to download / upload minor changes either locally or within the cloud...
If so is there a switch or can I request a feature change for a switch like --ignore-otherdirectories or similar ?

Sorry technically more than 1 question there ;)

Cheers
Bob

**Reply by abraunegg** [reply]
@bobmclane 
> So I think I have it working but if I have TB's of data in OneDrive and my "folder1" is in the root along with other files and folders in the root if I perform a --single-directory sync on "folder1" is it still checking every other file/folder in my OneDrive as it seems so?

The OneDrive API still sends *all* changes that you have made online (via the web interface, via other means). The `--single-directory` directive then filters out changes that are not in scope based on your application command request. You cannot tell the API to just send me *these* changes - it has to be done client side.

> Also will it do this every time as this could take me ages to download / upload minor changes either locally or within the cloud...

If you are using `--resync` or using 'sync_list' then yes. If not using either of these two, the *first* sync will setup a delta call, thus subsequent calls to the OneDrive API will be quicker.

> If so is there a switch or can I request a feature change for a switch like --ignore-otherdirectories or similar ?

This is what `--single-directory` does .. it ignores / filters out non-desired changes on OneDrive

**Reply by bobmclane** [reply]
@abraunegg - Hi is there any answer for the above?  If I leave it running a delta sync it will take seconds to upload the test docs I have in there but around 30 mins to download a test doc from OneDrive so I presume its scanning everything for changes even though I explicitly name just "folder1", is there a way to avoid this?
Thanks
Bob

**Reply by bobmclane** [reply]
Sorry ignore this last question, didnt see the reply was hidden... 
So I'm not running a re-sync and removed the sync_list but each time it still takes 30 mins just using this command -
onedrive --synchronize --single-directory 'folder1'
So are you saying that if other changes are occurring in other directories on OneDrive (from other machines) that even though we explicitly just specify "folder1" it will go thru all those changes anyway but just ignore synching them?

Thanks for all the help!
Cheers
Bob

**Reply by abraunegg** [reply]
This is because you are using the `--synchronize` command.

If you have a constant sync running with `--monitor`, the experience will be different because the delta position will be retained and used.

What you are currently doing is a 1 time sync, which, checks things then next time starts from scratch again.

**Reply by bobmclane** [reply]
Ah awesome thats working perfectly I'll set that up as a service... Thanks for all your help @abraunegg nice to have a OneDrive app on my Linux systems as well as Windows 👍 
Regards
Bob
